services/py-api/tests/unit_tests/test_service_layer/test_participant_registration_service.py

```diff
@@ -170,3 +170,124 @@ async def test_register_admin_participant_order_of_operations(
     # Check that the result is an `Err` of type HackathonCapacityExceededError
     assert isinstance(result, Err)
     assert isinstance(result.err_value, HackathonCapacityExceededError)
+
+@pytest.mark.asyncio
+async def test_register_random_participant_success(
+    p_reg_service: ParticipantRegistrationService,
+    hackathon_service_mock: Mock,
+    participant_repo_mock: Mock,
+    mock_input_data_random: ParticipantRequestBody,
+) -> None:
+    # Mock not full hackathon
+    hackathon_service_mock.check_capacity_register_random_participant_case = AsyncMock(return_value=True)
+
+    # Mock successful `create` responses for team and participant. These are the operations inside the passed callback
+    # to with_transaction
+    participant_repo_mock.create.return_value = Participant(
+        name=mock_input_data_random.name,
+        email=mock_input_data_random.email,
+        is_admin=False,
+        team_id=None,
+    )
+
+    hackathon_service_mock.create_random_participant.return_value = Ok(participant_repo_mock.create.return_value)
```

Comment:
Doesn't `create_random_participant` return `Result[Tuple[Participant, None], DuplicateEmailError | Exception]` and not a single OK value? Maybe change this to `Ok(
        (participant_repo_mock.create.return_value, None)
    )